elements/howto-menu/howto-menu.js

```diff
@@ -0,0 +1,216 @@
+/*
+ * A menu is a widget that offers a list of choices to the user,
+ * such as a set of actions or functions. A menu is usually opened,
+ * or made visible, by activating a menu button, choosing an item in a menu
+ * that opens a sub menu, or by invoking a command, such as Shift + F10 in
+ * Windows, that opens a context specific menu.
+ *
+ * The element that opens the menu is referenced with aria-labelledby.
+ * TODO(devnook): Should it be aria-labelledby or aria-describedby?
+ *
+ * See: https://www.w3.org/TR/wai-aria-practices-1.1/#menu
+ */
+(function() {
+  /**
+   * Define key codes to help with handling keyboard events.
+   */
+  const KEYCODE = {
+    DOWN: 40,
+    UP: 38,
+    ESCAPE: 27,
+  };
+
+  class HowtoMenu extends HTMLElement {
+
+    /**
+     * The constructor does work that needs to be executed _exactly_ once.
+     */
+    constructor() {
+      super();
+      this._handleBlur = this._handleBlur.bind(this);
+      this._opener = document.getElementById(
+         this.getAttribute('aria-labelledby'));
+    }
+
+    static get observedAttributes() {
+      return ['aria-hidden'];
```

Comment:
I'd suggest creating a boolean property/attribute pair called `.opened` and `[opened]`.

Using `expanded` makes me think it will also set `aria-expanded` but I don't think we need to do that here. Instead we could reserve `expanded`/`aria-expanded` just for submenu items.